### ISSUE TITLE
SceneKit API

### ISSUE THREAD

**Issue by scj643** [omz/Pythonista-Issues#241]
# Idea
An API just like the UI module except for https://developer.apple.com/reference/scenekit
Also https://developer.apple.com/reference/modelio

# Implementation
## Have object class that has...
* all the primitive shapes (cube, cone, ect.)
* Model IO objects

## Handling
Objects are treated like UI views with one layer of depth (you can't have an object within an object)

## Physics
If possible physics would be an option as well

# Reasoning
Being able to work with 3D objects would be awesome.

# Needs
Has to be a separate framework that has to be explicitly loaded to minimize impact.

**Comment by controversial** [comment]
There was a `sk` module for a while in one of the betas. It was taken out because it was buggy somehow. Don't remember details. This would indeed be useful, though.

**Comment by zrzka** [comment]
@omz isn't this covered with `scene`? I don't use SceneKit, but this issue is kind of broad. Would close it and specific requests can be filled per issue if something is missing / doesn't work.

**Comment by omz** [comment]
@zrzka No, SceneKit is a 3D API, while Pythonista's `scene` module (named before SceneKit existed on iOS) is 2D-only, and more similar to SpriteKit.

**Comment by zrzka** [comment]
@omz ah, okay, thanks for the explanation, it makes sense now.
